client/landing/stepper/declarative-flow/internals/steps-repository/domains/index.tsx

```diff
@@ -471,6 +473,7 @@ const DomainsStep: Step = function DomainsStep( { navigation, flow } ) {
 				</div>
 			}
 			recordTracksEvent={ recordTracksEvent }
+			goBack={ () => submit?.( { previousStep: true } ) }
```

Comment:
@kozer is right and `goBack` is deprecated. What about using `exitFlow` ?
The logic would live in `useStepNavigation` but it would be more clear that we are leaving the flow 👌 .  Everybody wins 😎 .

```suggestion
			goBack={ () => exitFlow?.( '/sites' ) }
```

We would also need to declare the `exitFlow` function in `declarative-flow/copy-site.tsx`:
```diff
diff --git a/client/landing/stepper/declarative-flow/copy-site.tsx b/client/landing/stepper/declarative-flow/copy-site.tsx
index 0be6d8983b..d279612cb9 100644
--- a/client/landing/stepper/declarative-flow/copy-site.tsx
+++ b/client/landing/stepper/declarative-flow/copy-site.tsx
@@ -80,3 +79,0 @@ const copySite: Flow = {
-					if ( providedDependencies.previousStep === true ) {
-						return window.location.assign( `/sites` );
-					}
@@ -136 +133,5 @@ const copySite: Flow = {
-		return { goNext, goBack, goToStep, submit };
+		const exitFlow = ( location: string ) => {
+			window.location.assign( location );
+		};
+
+		return { goNext, goBack, goToStep, submit, exitFlow };
```

Reply:
This is like we reinvented the `goBack` functionality.
Docs mention the following, so I'm a bit skeptical about this solution:

	 * @deprecated Avoid this method. Use submit() instead.
	 * If you need to navigate back and forth between
	 * stepper screens, consider adding screen(s) to a new
	 * stepper flow and linking directly between flows/screens.

Reply:
resolving this conversation since we decided to merge the changes and use `exitFlow`